packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -190,7 +213,8 @@ export class Alarm extends AlarmBase {
 
       // Evaluation
       comparisonOperator,
-      threshold: props.threshold,
+      threshold: isAnomalyDetection ? undefined : props.threshold,
```

Comment:
What will happen if it's anomaly detection and threshold value is passed in? Does it ignore the value or throw an error?

Reply:
If the comparison operator is for an anomaly detection alarm, there will be an error when CloudFormation will try to execute the template. Because providing a threshold for anomaly detection is not valid.

Reply:
Can we do this validation on CDK side to preemptively throw?